lib/ingestor-api/runtime/src/main.py

```diff
@@ -123,3 +125,9 @@ def who_am_i(username=Depends(dependencies.get_username)):
     Return username for the provided request
     """
     return {"username": username}
+
+
+# exception handling
+@app.exception_handler(RequestValidationError)
+async def validation_exception_handler(request, exc):
+    return JSONResponse(str(exc), status_code=422)
```

Comment:
What's the backstory for this change?

Can you provide an example of the new error format compared to the old format?  My hunch is that this makes it so that we are returning a raw string to the user?

Reply:
sorry I should have provided more context. It's an improvement they made in the VEDA version of the ingestor. My understanding is that it makes sure to return validation information when there is a validation error. 

For example, the collection endpoint. If you pass an invalid collection json (invalid with respect to the model described in schemas), then you will see an error indicating what's missing etc.

As far as I experienced, it's not the case without this handler. You are right that with this we're returning a raw string to the user.

Reply:
@alukach here the change in practice. Say I have a collection JSON that doesn't have a `links` field, and I submit that to the collection endpoint. 

I thought their change was actually providing more information but it's not the case, it just changes the format of the error. I think it's more readable, so that would be the improvement of this change. Let me know if you think it worths the addition. 

Client side error before the change : 

```
Error inserting collection.
{"detail":[{"loc":["body","links"],"msg":"field required","type":"value_error.missing"}]}
Traceback (most recent call last):
  File "/Users/emiletenezakis/devseed/maap-pgstac/test_insert_collection.py", line 145, in <module>
    response.raise_for_status()
  File "/Users/emiletenezakis/mambaforge/lib/python3.10/site-packages/requests/models.py", line 1021, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 422 Client Error: Unprocessable Entity for url: http://127.0.0.1:8000/collections?provided_by=emile+tenezakis
```

Client side error after the change : 

```
Error inserting collection.
"1 validation error for Request\nbody -> links\n  field required (type=value_error.missing)"
Traceback (most recent call last):
  File "/Users/emiletenezakis/devseed/maap-pgstac/test_insert_collection.py", line 145, in <module>
    response.raise_for_status()
  File "/Users/emiletenezakis/mambaforge/lib/python3.10/site-packages/requests/models.py", line 1021, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 422 Client Error: Unprocessable Entity for url: http://127.0.0.1:8000/collections?provided_by=emile+tenezakis
```

Reply:
To be honest, I'm not sure if this is something we should port over.  This change makes the API response more _human_ readable, but more difficult to parse via machine.  In my experience, APIs typically don't serve humans and instead serve client applications.  Building logic into a frontend client that highlights the field in the form that caused the validation error would be considerably more difficult to categorize and render when we're returning plaintext response.

Reply:
I see what you're saying, and I agree.